fetch.bs

```diff
@@ -2353,6 +2457,310 @@ X-Content-Type-Options           = "nosniff" ; case-insensitive</pre>
+ <li><p>Let <var>keyDomainName</var> be null.
+
+ <li><p>If <var>tokenBindingOrigin</var>'s <a for=origin>host</a> is an <a>IPv4 address</a> or an <a>IPv6 address</a>,
+ set <var>keyDomainName</var> to <var>tokenBindingOrigin</var>'s <a for=origin>host</a>.
+
+ <p>Otherwise, set <var>keyDomainName</var> to the
+ <dfn id=concept-registrable-domain/>registrable domain</dfn> of
+ <var>tokenBindingOrigin</var>'s host, determined by running the steps in the Algorithm section of
+ <a href=https://publicsuffix.org/list/>Public Suffix List</a>
+ [[PUBLIC-SUFFIX]]
+
+ <li><p>Let <var>tokenBindingKeyPair</var> be null.
+
+ <li><p>Set <var>tokenBindingKeyPair</var> to the result of looking up the
+ <a for=/>token-binding key</a> pair for <var>keyDomainName</var> and
+ <var>tokenBindingKeyParameters</var> in the <a for=/>token-binding key store</a>.
+
+ <li><p>If <var>tokenBindingKeyPair</var> is null, create a new key pair for
+ <var>keyDomainName</var> using <var>tokenBindingKeyParameters</var>, save it in the
+ <a for=/>token-binding key store</a>, and set <var>tokenBindingKeyPair</var> to that new key pair.
+
+ <li><p>Return <var>tokenBindingKeyPair</var>.
+</ol>
+
+<h4 id=computing-token-binding-id>Computing a Token Binding Id</h4>
+<p>The structure of a <a for=/>token binding ID</a> is described in
+ <a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3.2">section 3.2</a>
+ of [[!TOKBIND-PROTOCOL]].
+
+<p><dfn abstract-op export id=concept-compute-token-binding-id>Compute a Token Binding ID</dfn>
+using a <a for=/>token-binding key</a> <code>tokenBindingKeyPair</code>
+and <a for=connection>token-binding key parameters</a>
+<code>tokenBindingKeyParameters</code>, by running these substeps:
+
+<ol>
+ <li><p>Let <var>publicKeyData</var> be a <code>Uint8Array</code> object
+ containing the public key of <a for=/>token-binding key</a> pair
+ <code>tokenBindingKeyPair</code> and
+ <var>tokenBindingIdBytes</var> be an empty byte sequence.
+
+ <li><p>Append <code>tokenBindingKeyParameters</code> to
+ <var>tokenBindingIdBytes</var>.
+
+ <li><p>Append <var>publicKeyData</var>'s length (in network byte order) to <var>tokenBindingIdBytes</var>.
+
+ <li><p>Append <var>publicKeyData</var> to <var>tokenBindingIdBytes</var>.
+
+ <li><p>Return <var>tokenBindingIdBytes</var>.
+</ol>
+
+<h4 id=computing-token-binding-header-value>Computing a Token Binding Header Value</h4>
+
+<p><dfn abstract-op export id=concept-compute-token-binding-header-value>Compute a Token Binding header value</dfn>
+that can be passed in a `<a http-header><code>Sec-Token-Binding</code></a>`
+<a for=/>header</a> for a <a for=/>request</a>
+<var>httpRequest</var> and a <a>connection</a> <var>tlsConnection</var>,
+using the <a for=/>token-binding key store</a> and the
+<a>connection pool</a>, by running these substeps:
+
+<ol>
+ <li><p>Return null if any of the following are true.
+
+ <ol>
+  <li><p><var>httpRequest</var>'s <a for=request>use-token-binding flag</a> is not set.
```

Comment:
The "Computing a Token Binding Header Value" routine defined here is only called in one place, which already checked the use-token-binding flag. This check could probably be left out here, but I also don't mind a belt and suspenders check.

Reply:
Leaving as is.